src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt

```diff
@@ -0,0 +1,137 @@
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.BDDMockito.given
+import org.mockito.BDDMockito.then
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.client.CommunityApiClient
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.Address
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.AddressStatus
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.AllOffenderDetailsResponse
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.ContactDetails
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.OffenderManager
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.ProviderEmployee
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.Staff
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.Team
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.TrustOfficer
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.exception.PersonNotFoundException
+import java.time.LocalDate
+
+@ExtendWith(MockitoExtension::class)
+@ExperimentalCoroutinesApi
+class CaseSummaryOverviewServiceTest {
+
+  private lateinit var caseSummaryOverviewService: CaseSummaryOverviewService
+
+  @Mock
+  private lateinit var communityApiClient: CommunityApiClient
+
+  @BeforeEach
+  fun setup() {
+    caseSummaryOverviewService = CaseSummaryOverviewService(communityApiClient)
+  }
+
+  @Test
+  fun `throws exception when no person details available`() {
+    runBlockingTest {
+      given(communityApiClient.getAllOffenderDetails(anyString()))
+        .willReturn(Mono.empty())
+
+      val nonExistentCrn = "this person doesn't exist"
+      assertThatThrownBy {
+        runBlockingTest {
+          caseSummaryOverviewService.getOverview(nonExistentCrn)
+        }
+      }.isInstanceOf(PersonNotFoundException::class.java)
+        .hasMessage("No details available for crn: $nonExistentCrn")
+
+      then(communityApiClient).should().getAllOffenderDetails(nonExistentCrn)
+    }
+  }
+
+  @Test
+  fun `retrieves case summary when no offences available`() {
+    runBlockingTest {
+      val crn = "my wonderful crn"
+      given(communityApiClient.getAllOffenderDetails(anyString()))
+        .willReturn(Mono.fromCallable { allOffenderDetailsResponse })
+      given(communityApiClient.getConvictions(anyString()))
+        .willReturn(Mono.empty())
+
+      val response = caseSummaryOverviewService.getOverview(crn)
+
+      val personalDetails = response.personDetails!!
+      val offencesShouldBeEmpty = response.offences
+
+      assertThat(personalDetails.crn).isEqualTo(crn)
+      assertThat(personalDetails.age).isEqualTo(39)
```

Comment:
This age will break when the person becomes a year older. I've tried to handle this in similar way here: https://github.com/ministryofjustice/make-recall-decision-api/pull/62/files#diff-68e948597daf542f1164ac2e0971005beb2490eae21919aa41be449f5e38e48cR142

Reply:
Done